### DISCUSSION TITLE
Associate and Dissociate Tokens via Facade Contract

### DISCUSSION THREAD

**Discussion by mgarbs** [hiero-ledger/hiero-improvement-proposals#722] [Ideas]
HIP proposal to add associate and dissociate functionality to the proxy facade contract for tokens.

**Comment by Nana-EC** [comment]
Hey @mshakeg
I'd like to add you the working group for HIP 719
Your [suggestion](https://github.com/hashgraph/hedera-services/issues/5950#issuecomment-1570095335) for `isAssociate()` was incorporated into the design.
Although not yet implemented it will be and want to capture your influence here.
If that's okay let me know and please give me the right ~email address (since we can use github handle)~ name to note on the HIP

**Reply by mshakeg** [reply]
Hey @Nana-EC sure, you can use my full name: Mohammed Shaikjee

**Reply by Nana-EC** [reply]
@mgarbs can we please update the working group for HIP 719 with these details.
Thanks again for the contribution @mshakeg 
Thanks

**Reply by Nana-EC** [reply]
@mshakeg better late than never.
Your suggestion for `isAssociated()` is now in mainnet with 0.53 🙌🏾

**Reply by mshakeg** [reply]
@Nana-EC y'all didn't actually implement it the way I suggested where the account address is accepted as a parameter

> function isAssociated(address account) view external returns (bool associated);

**Reply by Nana-EC** [reply]
Hey @mshakeg,
This is unfortunate that the reviews didn't pick this up.
Would you be open to creating a new HIP with this request?
You can follow the structure of HIP 719, if not @lukelee-sl or I can carve out some time to get this going.
It would require approval from Heiro TSC and then Hedera Tech Com after that before implementation

**Reply by mshakeg** [reply]
Hey @Nana-EC unfortunately I don't have the time atm.

**Reply by Nana-EC** [reply]
Okay, will carve out some time to get a new HIP to address your original request
